### PR TITLE
Feat: guide to trigger workflows

### LINKED ISSUE #104: Document BPMN custom response
The custom response is a way to get feedback when triggering BPMN workflows from the API.  Very useful for debugging . Related to #103

### PR DESCRIPTION
**Pull request by MattDodsonEnglish**
Closes #104 in the process

### CODE REVIEW

File: content/how-to/bpmn/trigger-workflow.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+When a workflow has the default start event, trigger it through an API call.
+For example, an API trigger may originate from a custom frontend application or a developer's local machine.
+
+### Synchronous and asynchronous API triggers
+
+To start BPMN workflows, Rhize has two API operations:
+- `createAndRunBPMNSync` starts a workflow and waits for the process to complete or abort (synchronous).
+- `createAndRunBpmn` starts a workflow and does not wait for the response (asynchronous).
+
+The synchronous call offers slightly better debugging and does not execute further operations until it receives a response from the Rhize server.
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:30]:
```suggestion
The synchronous call does not execute further operations until it receives a response from the Rhize server.
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    jobState
+    customResponse
+  }
+}
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+The responses for these calls have two differences:
+- For synchronous calls, the returned `JobState` should be a finished value (such as `COMPLETED` or `ABORTED`). For asynchronous calls, the status is likely in progress.  a different value for synchronous calls.
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:63]:
```suggestion
- For synchronous calls, the returned `JobState` should be a finished value (such as `COMPLETED` or `ABORTED`). Asynchronous calls likely return an in-progress status, such as `RUNNING`.
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+### `CustomResponse`
+
+The `customResponse` is a special variable to return data in the response to clients that run `createAndRunBPMNSync` operations.
+
+You can set the `customResponse` in any [element]({{< relref "/how-to/bpmn/bpmn-elements" >}}) that has an `Output` or `Input response` parameter.
+It can use any data from the {{< abbr "process variable context" >}}, including variables added on the fly.
+Its format can use JSON or JSONata.
+
+Functionally, only the last value of the `CustomResponse` is returned to the client that sent the response.
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:74]:
```suggestion
Functionally, only the last value of the `customResponse` is returned to the client that sent the response.
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  "createAndRunBpmnSyncId": "API_demoCallToRemoteAPI",
+  "version": "3"
+}
+```
+
+## Start from a message
+
+The [message start event]({{< relref "/how-to/bpmn/bpmn-elements#message-start-event" >}}) subscribes to a topic on the Rhize broker.
+Whenever a message is published to this topic, the workflow is triggered.
+The Rhize broker can receive messages in MQTT, NATS, and OPC UA.
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:114]:
```suggestion
The Rhize broker can receive messages published over MQTT, NATS, and OPC UA.
```

**Review by MattDodsonEnglish — COMMENTED** [review]
Text clean up

File: content/how-to/bpmn/trigger-workflow.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+- For synchronous calls, the returned `JobState` should be a finished value (such as `COMPLETED` or `ABORTED`). Asynchronous calls likely return an in-progress status, such as `RUNNING`.
+- Only the synchronous call receives data in the `customResponse`. For details, refer to the next section.
+
+### `CustomResponse`
+
+The `customResponse` is a special variable to return data in the response to clients that run `createAndRunBPMNSync` operations.
+
+You can set the `customResponse` in any [element]({{< relref "/how-to/bpmn/bpmn-elements" >}}) that has an `Output` or `Input response` parameter.
+It can use any data from the {{< abbr "process variable context" >}}, including variables added on the fly.
+Its format can use JSON or JSONata.
```
Comment (Jarrah-libremfg) [inline content/how-to/bpmn/trigger-workflow.md:72]: The format will always be JSON. Some part of the process must create a variable in the context called `customResponse` and that variable is returned as a JSON string.

File: content/how-to/bpmn/trigger-workflow.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+To add variables, use the `variables` input argument.
+Note that the variables are accessible from their root object.
+For example, a workflow would access the following string value at `$.input.message`:
+
+```gql{
+"variables": "{\"input\":{\"message\":\"CORRECT\"}}"
+}
+```
+
+To specify a version, use the `version` property. For example, this input specifies that Rhize run version `3` of the `API_demoCallToRemoteAPI` workflow:
```
Comment (Jarrah-libremfg) [inline content/how-to/bpmn/trigger-workflow.md:101]: Probably worth adding that if no version is specified the "active version" (if any) is executed.

File: content/how-to/bpmn/trigger-workflow.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+<!-- vale off -->
+{{% bigFigure 
+alt="BPMN message start with conditional evaluation"
+src="/images/bpmn/rhize-bpmn-message-start-throw-conditional.png"
+width="65%"
+caption="Download this workflow as a [BPMN template](https://github.com/libremfg/bpmn-templates/tree/main/msg-start-and-throw)."
+ %}}
+<!-- vale on -->
+
+Note that, for a workflow to run from a message start event, the workflow **must be enabled.**
```
Comment (Jarrah-libremfg) [inline content/how-to/bpmn/trigger-workflow.md:130]: Not sure if you want to go to this level of detail but there are two things:
1. The workflow needs an active version.
2. The WorkflowMessage of that active version needs to be enabled.

File: content/how-to/bpmn/trigger-workflow.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    customResponse
+  }
+}
+```
+{{% /tab %}}
+
+{{% tab "Async" %}}
+```gql
+mutation AsyncCall{
+  createAndRunBPMNSync{
```
[the remark this answers is not in the document]
Comment (MattDodsonEnglish) [reply content/how-to/bpmn/trigger-workflow.md:52]: > "each operation is almost identical

More about the call syntax, but I'm going to make it more explicit.

> The call syntax for both operations are similar.

File: content/how-to/bpmn/trigger-workflow.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+### Synchronous and asynchronous API triggers
+
+To start BPMN workflows, Rhize has two API operations:
+- `createAndRunBPMNSync` starts a workflow and waits for the process to complete or abort (synchronous).
+- `createAndRunBpmn` starts a workflow and does not wait for the response (asynchronous).
+
+The synchronous call does not execute further operations until it receives a response from the Rhize server.
+On the other hand, the asynchronous operation frees up the call system to do more work, no matter whether the workflow runs correctly.
+
+The call syntax for the each operation is almost identical.
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:33]:
```suggestion
Both operations have  similar call syntax.
```

File: content/how-to/bpmn/trigger-workflow.md (hunk 5 above)
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:72]:
```suggestion
```

File: content/how-to/bpmn/trigger-workflow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (Jarrah-libremfg) [reply content/how-to/bpmn/trigger-workflow.md:52]: I actually had no issue with the text above, just the missing arguments in this mutation.

File: content/how-to/bpmn/trigger-workflow.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+To add variables, use the `variables` input argument.
+Note that the variables are accessible from their root object.
+For example, a workflow would access the following string value at `$.input.message`:
+
+```gql{
+"variables": "{\"input\":{\"message\":\"CORRECT\"}}"
+}
+```
+
+To specify a version, use the `version` property. For example, this input specifies that Rhize run version `3` of the `API_demoCallToRemoteAPI` workflow:
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:100]:
```suggestion
To specify a version, use the `version` property. For example, this input instructs Rhize to run version `3` of the `API_demoCallToRemoteAPI` workflow:
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+When a workflow has the default start event, trigger it through an API call.
+For example, an API trigger may originate from a custom frontend application or a developer's local machine.
+
+### Synchronous and asynchronous API triggers
+
+To start BPMN workflows, Rhize has two API operations:
+- `createAndRunBPMNSync` starts a workflow and waits for the process to complete or abort (synchronous).
+- `createAndRunBpmn` starts a workflow and does not wait for the response (asynchronous).
+
+The synchronous call does not execute further operations until it receives a response from the Rhize server.
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:30]:
```suggestion
Use the synchronous operation if you want to receive information about the result of the workflow in the call response.
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
```
[the remark this answers is not in the document]
Comment (MattDodsonEnglish) [reply content/how-to/bpmn/trigger-workflow.md:30]: > s. For an Async BPMN call you would need to read the JobResponse from NATS or set up storeJobResponse correctly.

This is a really good point, but I don't want to make this particular guide too complex. However I have an issue  to create a guide to debug and error handle BPMNs, and this made me realize I need to put information about inspecting the workflow state through API there.

https://github.com/libremfg/libremfg.github.io/issues/103

File: content/how-to/bpmn/trigger-workflow.md [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+To specify a version, use the `version` property. For example, this input specifies that Rhize run version `3` of the `API_demoCallToRemoteAPI` workflow:
+
+```json
+{
+  "createAndRunBpmnSyncId": "API_demoCallToRemoteAPI",
+  "version": "3"
+}
+```
+
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:108]:
```suggestion

```

If the `version` property is empty, Rhize runs the active version of the workflow (if one exists).

File: content/how-to/bpmn/trigger-workflow.md (hunk 7 above)
Comment (MattDodsonEnglish) [reply content/how-to/bpmn/trigger-workflow.md:129]: I just checked and I think the UI does not permit the User to enable a workflow if it's not active, so for now I'm going to leave that part out. Of course we can add it back in if users complain.

File: content/how-to/bpmn/trigger-workflow.md [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+    customResponse
+  }
+}
+```
+{{% /tab %}}
+
+{{% tab "Async" %}}
+```gql
+mutation AsyncCall{
+  createAndRunBPMNSync{
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:52]:
```suggestion
{{% tabs %}}
{{% tab "Synchronous" %}}
```gql
mutation SynchronousCall($createAndRunBpmnSyncId: String!, $version: String ) {
  createAndRunBpmnSync(id: $createAndRunBpmnSyncId, version: $version) {
    id
    jobState
    customResponse
  }
}
```
{{% /tab %}}

{{% tab "Async" %}}
```gql
mutation SynchronousCall{
  createAndRunBpmn(id: "API_demo_custom_response") {
    id
    jobState
    customResponse
  }
}
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 15]
```diff
@@ -0,0 +1,146 @@
...
+{{% tabs %}}
+{{% tab "Synchronous" %}}
+```gql
+mutation SynchronousCall($createAndRunBpmnSyncId: String!, $version: String ) {
+  createAndRunBpmnSync(id: $createAndRunBpmnSyncId, version: $version) {
+    id
+    jobState
+    customResponse
+  }
+}
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:45]:
```suggestion
mutation sychronousCall{
  createAndRunBpmnSync(id: "API_demo_custom_response") {
    id
    jobState
    customResponse
  }
}

```

File: content/how-to/bpmn/trigger-workflow.md (hunk 14 above)
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:52]:
```suggestion
mutation asyncCall{
  createAndRunBpmn(id: "API_demo_custom_response") {
    id
    jobState
    customResponse
  }
}
```

File: content/how-to/bpmn/trigger-workflow.md [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+{{% tab "Async" %}}
+```gql
+mutation AsyncCall{
+  createAndRunBPMNSync{
+    id
+    jobState
+    customResponse
+  }
+}
```
Comment (MattDodsonEnglish) [inline content/how-to/bpmn/trigger-workflow.md:57]:
```suggestion
```

**Pull request merged**